pyrokinetics/pyro.py

```diff
@@ -222,6 +229,9 @@ def write_gk_file(self,
 
         self.gk_code.write(self, file_name, directory=self.run_directory)
 
+        # Ensure that gk_code is unchanged on exit
+        self.gk_code = original_gk_code
```

Comment:
@d7919  Not sure if it is necessary to save the original gk_code. No data is lost by changing the code as the namelist data for each code is saved under a separate object.

For example the if we load in a GS2 input file and the namelist data with all the flags is saved under pyro.gs2_input. When changing code the pyro.gs2_input remains. Thoughts?

Reply:
I prefer function calls to not have side effects if possible. The original approach makes it clear that the user is changing the gk_code value of the pyro instance whilst this function call doesn't make it obvious so I thought we should revert to the original type.

Reply:
Fair enough, I think I agree with that

Reply:
This currently won't quite work as you can only set gk_code with a string. We need modify the gk_code.setter to handle gk_code objects as inputs

Reply:
That would work but we definitely need to re-vist how setting gk_code/gk_type works